optima/optimization.py

```diff
@@ -212,23 +212,22 @@ def constrainbudget(origbudget=None, budgetvec=None, totalbudget=None, budgetlim
 
     # Calculate the current total budget
     currenttotal = sum(constrainedbudget[:]) # WARNING, assumes it's an odict
-    scaleratio = totalbudget/float(currenttotal) # Calculate the ratio between the original budget and the supplied budget
```

Comment:
@cliffckerr I only changed `optimization.py` to add `stoppingfunc` in various places - the other changes might accidentally be reverting updates due to the order in which they were committed on my branch - can you please double check which is the correct version?